File: frontend/src/services/departmentService.ts
import api from "./api"

export interface DepartmentRequest { name: string; description?: string }
export interface DepartmentResponse { id: string; name: string; description?: string; employeeCount?: number; createdAt: string }

const departmentService = {
  getAll() { return api.get<DepartmentResponse[]>("/departments") },
  getById(id: string) { return api.get<DepartmentResponse>(`/departments/${id}`) },
  create(data: DepartmentRequest) { return api.post<DepartmentResponse>("/departments", data) },
  update(id: string, data: DepartmentRequest) { return api.put<DepartmentResponse>(`/departments/${id}`, data) },
  delete(id: string) { return api.delete(`/departments/${id}`) },
}

export default departmentService

File: backend/src/main/java/com/hrms/employee/service/DepartmentService.java
package com.hrms.employee.service;

import com.hrms.employee.dto.DepartmentRequest;
import com.hrms.employee.dto.DepartmentResponse;

import java.util.List;
import java.util.UUID;

public interface DepartmentService {
    DepartmentResponse create(DepartmentRequest request);
    DepartmentResponse getById(UUID id);
    List<DepartmentResponse> getAll();
    DepartmentResponse update(UUID id, DepartmentRequest request);
    void delete(UUID id);
}
